lib/server.js

```diff
@@ -24,6 +26,16 @@ const initServer = function(
     applications = apps.createAppsIndex(applications);
   }
 
+  // versions cached for efficient search when provided version is a range
+  this._cachedVersions = new Map();
+  applications.forEach((appVersions, appName) => {
+    const versions = Array.from(appVersions)
+      .filter(app => app[0] !== 'latest')
+      .map(version => [new semver.SemVer(version[0]), version[1]])
```

Comment:
Can you please use destructuring in the left-hand side of the lambda to give some descriptive names for these values (i.e., `([something, somethingElse]) => [new semver.SemVer(something), somethingElse]`)?

Aside from that, why do `app` and `version` identifiers refer to the same objects in this and previous lines?